src/App.jsx

```diff
@@ -14,33 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGHT = 'lenght';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getGoodsPrepared(goods, { sortField }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABETICALLY:
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case SORT_FIELD_LENGHT:
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+
+      default:
+        return 0;
+    }
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  let visibleGoods = getGoodsPrepared(goodsFromServer, { sortField });
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
```

Comment:
1. reversing is also a sorting type. So you can just move your `reverse` logic to `getGoodsPrepared`
2. https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
```suggestion
  const [isReversed, setIsReversed] = useState(false);
  const visibleGoods = getGoodsPrepared(goodsFromServer, { sortField, isReversed });
```